imap/lmtp_sieve.c

```diff
@@ -785,7 +785,14 @@ static int send_forward(sieve_redirect_context_t *rc,
 #endif
     }
     else {
-        smtp_envelope_add_rcpt(&sm_env, rc->addr);
+        struct address *addr = NULL;
+        char *rcpt = NULL;
+
+        parseaddr_list(rc->addr, &addr);
+        rcpt = address_get_all(addr, 1);
```

Comment:
Is this guaranteed to only ever be called with valid addr-spec values? I know that the redirect rule sieve parser validates the addr-spec, but could this be called from anywhere else where validation is not guaranteed?
If so, then a NULL check here should fix this.

Reply:
It shouldn't be called from anywhere else.  I'm not really using the parser here to validate the address (as you said, the script parser has already determined that its not junk), I'm just using it to break it down into its component parts to strip the phrase part.